File: graphql/src/main/java/ru/romanow/protocols/graphql/service/BuilderHelper.java
package ru.romanow.protocols.graphql.service;

import ru.romanow.protocols.graphql.domain.Author;
import ru.romanow.protocols.graphql.domain.Book;
import ru.romanow.protocols.graphql.model.AuthorResponse;
import ru.romanow.protocols.graphql.model.BookResponse;

public class BuilderHelper {
    public static AuthorResponse buildAuthorInfo(Author author) {
        return new AuthorResponse()
                .setId(author.getId())
                .setName(author.getName())
                .setAge(author.getAge())
                .setExperience(author.getExperience());
    }

    public static BookResponse buildBookInfo(Book book) {
        return new BookResponse()
                .setId(book.getId())
                .setIsn(book.getIsn())
                .setName(book.getName())
                .setPrice(book.getPrice())
                .setPageCount(book.getPageCount())
                .setAuthor(buildAuthorInfo(book.getAuthor()));
    }
}

File: graphql/src/main/java/ru/romanow/protocols/graphql/service/BuilderHelper.kt
package ru.romanow.protocols.graphql.service

import ru.romanow.protocols.graphql.domain.Author
import ru.romanow.protocols.graphql.domain.Book
import ru.romanow.protocols.graphql.model.AuthorResponse
import ru.romanow.protocols.graphql.model.BookResponse

fun buildAuthorInfo(author: Author) =
    AuthorResponse(
        id = author.id,
        name = author.name,
        age = author.age,
        experience = author.experience
    )

fun buildBookInfo(book: Book) =
    BookResponse(
        id = book.id,
        isn = book.isn,
        name = book.name,
        price = book.price,
        pageCount = book.pageCount,
        author = buildAuthorInfo(book.author!!)
    )